pkg/docker/docker.go

```diff
@@ -126,6 +126,62 @@ type RunContainerOptions struct {
 	CapDrop          []string
 }
 
+// asDockerConfig converts a RunContainerOptions into a Config understood by the
+// docker client library.
+func (rco RunContainerOptions) asDockerConfig() docker.Config {
+	return docker.Config{
+		Image: getImageName(rco.Image),
+		User:  rco.User,
+		// TODO why do we call runContainerTar with a "partial"
+		// docker.Config?
+
+		Env: rco.Env,
+
+		// TODO review how callers use these options to pass stdin.
+		// Probably they are not necessary, as they're related to
+		// attaching to the container (that is often unnecessary).
+		OpenStdin: rco.Stdin != nil,
+		StdinOnce: rco.Stdin != nil,
+		// TODO surprinsingly, we do not set AttachStdin...
+
+		AttachStdout: rco.Stdout != nil,
+	}
+}
+
+// asDockerHostConfig converts a RunContainerOptions into a HostConfig
+// understood by the docker client library.
+func (rco RunContainerOptions) asDockerHostConfig() docker.HostConfig {
+	hostConfig := docker.HostConfig{
+		CapDrop: rco.CapDrop,
+
+		PublishAllPorts: rco.TargetImage,
+		// TODO why didn't we always set NetworkMode?!
+		NetworkMode: rco.NetworkMode,
+	}
+	if rco.CGroupLimits != nil {
+		hostConfig.Memory = rco.CGroupLimits.MemoryLimitBytes
+		hostConfig.MemorySwap = rco.CGroupLimits.MemorySwap
+		hostConfig.CPUShares = rco.CGroupLimits.CPUShares
+		hostConfig.CPUQuota = rco.CGroupLimits.CPUQuota
+		hostConfig.CPUPeriod = rco.CGroupLimits.CPUPeriod
+	}
+	return hostConfig
+}
+
+// asDockerCreateContainerOptions converts a RunContainerOptions into a
+// CreateContainerOptions understood by the docker client library.
+func (rco RunContainerOptions) asDockerCreateContainerOptions() docker.CreateContainerOptions {
+	config := rco.asDockerConfig()
+	hostConfig := rco.asDockerHostConfig()
+	return docker.CreateContainerOptions{
+		// TODO we should probably be setting some S2I-related name
+		// instead.
```

Comment:
@bparees agree we should give the image an s2i related name? That was the decision we made in origin. Empty string means Docker will generate names like "fluffy_panda" :panda_face: